FlowCrypt/src/main/java/com/flowcrypt/email/jetpack/viewmodel/PrivateKeysViewModel.kt

```diff
@@ -389,16 +389,16 @@ class PrivateKeysViewModel(application: Application) : BaseNodeApiViewModel(appl
           throw IllegalStateException("Passphrase for key with longid $longId not found")
         }
 
-        val decryptedKey = PgpKey.decryptKey(nodeKeyDetails.privateKey!!, oldPassphrase!!)
-
-        if (TextUtils.isEmpty(decryptedKey)) {
-          throw IllegalStateException("Can't decrypt key with longid " + longId!!)
-        }
-
-        val encryptedKey = PgpKey.encryptKey(decryptedKey, newPassphrase)
-
-        if (TextUtils.isEmpty(encryptedKey)) {
-          throw IllegalStateException("Can't encrypt key with longid " + longId!!)
+        val encryptedKey: String
```

Comment:
```
val encryptedKey = try {
          PgpKey.changeKeyPassphrase(nodeKeyDetails.privateKey!!, oldPassphrase!!, newPassphrase)
        } catch (e: Exception) {
          throw IllegalStateException("Can't change passphrase for the key with longid "
              + longId!!, e)
        }
```

Reply:
@DenBond7 this is an interesting thing - after a lot of discussion between me, Ivan and Julian, we settled for the pattern that Ivan used on other repositories, and we disabled the IDE nudges to refactor it as you suggest.

The reason is that we'd like the code to be very understandable to people who only know Java and see Kotlin for the first time (which may be the case for the security reviewers for example, or for new team members).

And so instead of:

```kt
val x = if(cond) {
  1
} else {
  2
}
```

we'd write

```
val x: Int
if(cond) {
  x = 1
} else {
  x = 2
}
```

It's more code but we think it increases clarity. 

In situations when you really want succinct usage, we can use `when` which is I think better structured for implicit returns:

```kt
val x: Int
val x = when (cond) {
  true -> x,
  false -> y
}
```

I think that is clearer than if or try/catch with implicit returns.

Similarly with try/catch. Generally also trying to avoid implicit returns in long lambdas, like this:

```diff
whatever(something, lambda = {
  something()
  else()
-  thisIsReturnedButItsNotObvious()
+  return@lambda thisIsClearer()
})
```

We have never discussed code style much on Android because you were the only contributor and the code style was already sensible. However as we have several people who may contribute to Kotlin code these days, we should use relatively consistent styling across repositories.

(I'll file an issue with this comment)

Reply:
@tomholub I'm marking it as resolved